src/openforms/submissions/attachments.py

```diff
@@ -83,17 +85,94 @@ def append_file_num_postfix(
     return f"{new_name}{postfix}{ext}"
 
 
+@dataclass
+class UploadContext:
+    upload: TemporaryFileUpload
+    component: Component
+    index: int  # one-based
+    form_key: str  # formio key
+    num_uploads: int
+
+
+def iter_step_uploads(
+    submission_step: SubmissionStep, data=None
+) -> Iterator[UploadContext]:
+    """
+    Iterate over all the uploads for a given submission step.
+
+    Yield every uploaded file and its context within the form step for further
+    processing.
+    """
+    data = data or submission_step.data
+    components = list(submission_step.form_step.iter_components(recursive=True))
+    uploads = resolve_uploads_from_data(components, data)
+    for key, (component, upload_instances) in uploads.items():
+        # formio sends a list of uploads even with multiple=False
+        for i, upload in enumerate(upload_instances, start=1):
+            yield UploadContext(
+                upload=upload,
+                form_key=key,
+                component=component,
+                index=i,
+                num_uploads=len(upload_instances),
+            )
+
+
+def validate_uploads(submission_step: SubmissionStep, data: Optional[dict]) -> None:
+    """
+    Validate the file uploads in the
```

Comment:
In the... The suspense is killing me ;-)

Reply:
you will never know :-)